Sources/SyncDataProviders/Credentials/internal/SyncableCredentialsAdapter.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  SyncableCredentialsAdapter.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import BrowserServicesKit
+import DDGSync
+import Foundation
+
+struct SyncableCredentialsAdapter {
+
+    let syncable: Syncable
+
+    init(syncable: Syncable) {
+        self.syncable = syncable
+    }
+
+    var uuid: String? {
+        syncable.payload["id"] as? String
+    }
+
+    var isDeleted: Bool {
+        syncable.isDeleted
+    }
+
+    var encryptedDomain: String? {
+        syncable.payload["domain"] as? String
+    }
+
+    var encryptedTitle: String? {
+        syncable.payload["title"] as? String
+    }
+
+    var encryptedUsername: String? {
+        syncable.payload["username"] as? String
+    }
+
+    var encryptedPassword: String? {
+        syncable.payload["password"] as? String
+    }
+
+    var encryptedNotes: String? {
+        syncable.payload["notes"] as? String
+    }
+}
+
+extension Syncable {
+
+    init(syncableCredentials: SecureVaultModels.SyncableCredentials, encryptedUsing encrypt: (String) throws -> String) throws {
+        var payload: [String: Any] = [:]
+
+        payload["id"] = syncableCredentials.metadata.uuid
+
+        guard let credential = syncableCredentials.credentials else {
+            payload["deleted"] = ""
+            self.init(jsonObject: payload)
+            return
+        }
+
+        print("Syncable init \(syncableCredentials.metadata.uuid)")
```

Comment:
Remove or replace with log?

Reply:
I'm removing this. It was here purely for debugging. Early days, where it was taking 1s between two subsequent inits 😬